src/Build/Definition/ProjectCollection.cs

```diff
@@ -309,7 +309,15 @@ public ProjectCollection(IDictionary<string, string> globalProperties, IEnumerab
             _loadedProjects = new LoadedProjectCollection();
             ToolsetLocations = toolsetDefinitionLocations;
             MaxNodeCount = maxNodeCount;
-            ProjectRootElementCache = new ProjectRootElementCache(false /* do not automatically reload changed files from disk */, loadProjectsReadOnly);
+
+            if (Traits.Instance.UseSimpleProjectRootElementCacheConcurrency)
```

Comment:
Add a test to ProjectCollection to assert that setting this environment property changes affects the cache type.